rotkehlchen/balances/historical.py

```diff
@@ -75,18 +78,56 @@ def get_asset_balance(self, asset: Asset, timestamp: Timestamp) -> HistoricalBal
 
         return {'amount': amounts[asset], 'price': price}
 
+    def get_asset_amounts(
+            self,
+            asset: Asset,
+            from_ts: Timestamp,
+            to_ts: Timestamp,
+    ) -> dict[Timestamp, FVal]:
+        """Get historical balance amounts for the given asset within the given time range.
+
+        It does not include the value of the balances in the user's profit currency as
+        that is handled separately by the frontend in a different request.
+
+        May raise:
+        - NotFoundError if no events exist for the asset in the specified time period.
+        """
+        events, _ = self._get_events_and_currency(from_ts=from_ts, to_ts=to_ts, assets=(asset,))
+        if len(events) == 0:
+            raise NotFoundError(f'No historical data found for {asset} within {from_ts=} and {to_ts=}.')  # noqa: E501
+
+        total_balance = ZERO
+        amounts: dict[Timestamp, FVal] = defaultdict(lambda: ZERO)
+        for event in events:
+            if (
+                (direction := event.maybe_get_direction()) is None or
+                direction == EventDirection.NEUTRAL
+            ):
+                continue
+
+            if direction == EventDirection.IN:
+                total_balance += event.balance.amount
+            else:  # EventDirection.OUT
+                total_balance -= event.balance.amount
```

Comment:
I also think here we need a communication with the frontend via a websocket. If the `total_amount` ever becomes negative that means that either events are missing or that an event was not decoded correctly.

This can be done in a different PR, but needs to be done.